### PR TITLE
feat(sumologic-mock): use shorter name for serviceaccount

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Add support for setup job by using helm-hooks. This is to run Sumo Logic Mock before setup job, so it can use it as terraform api endpoint

I don't like when I start doing something and then I see that there are some fixes to do. So ended up with 3 changelog entries

### Checklist

- [ ] Changelog updated or skip changelog label added
- [ ] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

File: deploy/helm/sumologic/templates/_helpers/_common.tpl [hunk 1]
```diff
@@ -608,5 +616,5 @@ Example usage:
 {{- end -}}
 
 {{- define "sumologic.sumologic-mock.name.roles.serviceaccount" -}}
-{{- template "sumologic.fullname" . }}
+{{- template "sumologic.fullname" . }}-sumologic-mock
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/templates/_helpers/_common.tpl:619]: I think we already know this is a "Sumo Logic" mock. Also we have had issues with names being too long.
```suggestion
{{- template "sumologic.fullname" . }}-mock
```

Comment (astencel-sumo) [reply deploy/helm/sumologic/templates/_helpers/_common.tpl:619]: Oh I see the name `...-sumologic-mock` is already being used in `_sumologic_mock.tpl`, I'm apparently a bit late with this suggestion. Can we still change those names to be shorter?

Comment (sumo-drosiek) [reply deploy/helm/sumologic/templates/_helpers/_common.tpl:619]: sure, I will prepare it as separate PR to merge before this one

Comment (sumo-drosiek) [reply deploy/helm/sumologic/templates/_helpers/_common.tpl:619]: #3536

File: deploy/helm/sumologic/README.md [hunk 2]
```diff
@@ -502,6 +502,7 @@ The following table lists the configurable parameters of the Sumo Logic chart an
...
 | `debug.sumologicMock.image`                                                           | Docker image for the Sumo Logic Mock.                                                                                                                                                                                                                                                                      | `{"repository": "public.ecr.aws/sumologic/sumologic-mock", "tag": "2.22.0-59-g245ae92", "pullPolicy": "IfNotPresent"}`                                                               |
+| `debug.enableLocalMode`                                                               | Use it along with `debug.sumologicMock.enabled` in order to install collection locally without accessId, accessKey                                                                                                                                                                                         | `false`                                                                                                                                                                              |
```
Comment (aboguszewski-sumo) [inline deploy/helm/sumologic/README.md:505]: Will we want to add some options to configure the local mode? If so, then I'd make a key `debug.localMode.enable` instead.

Comment (sumo-drosiek) [reply deploy/helm/sumologic/README.md:505]: I do not plan any for now. For me it is just using mock for registration and collector creation process. Do you have any idea what could be change? Also as experimental and debugging feature we can change it any time :sweat:

Comment (astencel-sumo) [reply deploy/helm/sumologic/README.md]: Ideas to customize in the future:
- only send specific signal to mock (logs/metrics/traces)
- only send to mock vs. send to both mock and Sumo

File: tests/helm/testdata/goldenfile/cleanup/sumologic-mock-no-secret.output.yaml [hunk 3]
```diff
@@ -51,7 +51,7 @@ spec:
                 name: RELEASE-NAME-sumologic-setup
           env:
             - name: SUMOLOGIC_BASE_URL
-              value: http://RELEASE-NAME-sumologic-sumologic-mock.sumologic:3000/terraform/api/
+              value: http://RELEASE-NAME-sumologic-mock.sumologic:3000/terraform/api/
```
Comment (sumo-drosiek) [inline tests/helm/testdata/goldenfile/cleanup/sumologic-mock-no-secret.output.yaml:54]: I have no idea what happened here

**Review by astencel-sumo — APPROVED** [review]

File: deploy/helm/sumologic/README.md (hunk 2 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/README.md]: > only send to mock vs. send to both mock and Sumo

This is supported right now

> only send specific signal to mock (logs/metrics/traces)

This also (in scenario when both Sumo Logic and Mock are being used)

**Pull request merged**